core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -59,7 +59,15 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info) }
+    rddInfos.foreach { info =>
```

Comment:
It's likely to be slower IMO (as it performs a mutation every time, as opposed to only mutating when needed - the modified cache page will need to sent back to RAM at some point, resulting in extra traffic on the bus).